[cmake] for -DPLATFORM=DRM, unset CMP0054 causes (re)compilation failure

- [X] I tested it on latest raylib version from master branch [b0f3a2c2]
 - [X] I checked there is no similar issue already reported
 - [X] My code has no errors or misuse of raylib

### Issue description

If PLATFORM=DRM and cmake files are edited and library (re)compiled, "PLATFORM" and dependent variables do not propagate correctly.

This only happens with DRM because "DRM" is itself a variable in the CMakeCache.txt and because of old, crufty, confusing reasons, documented in CMP0054.

The problem is that when you run cmake the 1st time, it creates an entry in CMakeCache.txt CALLED "DRM"
```
$ grep ^DRM build/linux/CMakeCache.txt 
DRM:FILEPATH=/usr/lib/x86_64-linux-gnu/libdrm.so
```

When run the 2nd time, Line 68 of cmake/LibraryConfigurations.cmake
```
elseif (${PLATFORM} MATCHES "DRM")
```
Expands 1st to:
```
elseif(DRM MATCHES DRM)
```
But, due to CMP0054 OLD behavior, the token "DRM" is further expanded to the value 
```
elseif(/usr/lib/x86_64-linux-gnu/libdrm.so MATCHES DRM)
```
Which it obviously doesn't. 

### Environment

I have reproduced this problem on two systems:

- Raspberry Pi 64bit, Ubuntu 20.04 Jammy, cmake 3.22.1, Ninja
- x86_64, Ubuntu 20.04 Focal, cmake 3.23.2, GNU Makefiles

### Issue Screenshot

See code example...

### Code Example

Configure build:

```
> cmake -B build/linux -DPLATFORM=DRM 
[...]
-- Audio Backend: miniaudio
-- Building raylib static library
-- Generated build type: Debug
-- Compiling with the flags:
--   PLATFORM=PLATFORM_DRM            ### <---- PLATFORM propagated
--   GRAPHICS=GRAPHICS_API_OPENGL_ES2
-- Building examples is enabled
[...]
```
And build:
```
> cmake --build build/linux
# build succeeds
```

Now edit any cmake file, this causes the next build to re-check all it's variables. Simulate here with 'touch.'

```
> touch cmake/CompileDefinitions.cmake # anything that causes cmake to re-run
> cmake --build build/linux
cmake --build build/linux/
-- Testing if -Werror=pointer-arith can be used -- compiles
-- Testing if -Werror=implicit-function-declaration can be used -- compiles
-- Testing if -fno-strict-aliasing can be used -- compiles
[...]
-- Audio Backend: miniaudio
-- Building raylib static library
-- Generated build type: Debug
-- Compiling with the flags:
--   PLATFORM=                                      ##### <===== PLATFORM IS MISSING
--   GRAPHICS=GRAPHICS_API_OPENGL_33
-- Building examples is enabled
-- Testing if -std=c11 can be used -- compiles
CMake Warning (dev) at examples/CMakeLists.txt:110 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "DRM" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
This warning is for project developers.  Use -Wno-dev to suppress it.

[...]
/usr/bin/cmake -S/Volumes/sensitive/raylib -B/Volumes/sensitive/raylib/build/linux --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /Volumes/sensitive/raylib/build/linux/CMakeFiles /Volumes/sensitive/raylib/build/linux//CMakeFiles/progress.marks
/usr/bin/make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/Volumes/sensitive/raylib/build/linux'
/usr/bin/make  -f raylib/CMakeFiles/raylib.dir/build.make raylib/CMakeFiles/raylib.dir/depend
make[2]: Entering directory '/Volumes/sensitive/raylib/build/linux'
cd /Volumes/sensitive/raylib/build/linux && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /Volumes/sensitive/raylib /Volumes/sensitive/raylib/src /Volumes/sensitive/raylib/build/linux /Volumes/sensitive/raylib/build/linux/raylib /Volumes/sensitive/raylib/build/linux/raylib/CMakeFiles/raylib.dir/DependInfo.cmake --color=
Dependencies file "raylib/CMakeFiles/raylib.dir/rcore.c.o.d" is newer than depends file "/Volumes/sensitive/raylib/build/linux/raylib/CMakeFiles/raylib.dir/compiler_depend.internal".
Consolidate compiler generated dependencies of target raylib
make[2]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
/usr/bin/make  -f raylib/CMakeFiles/raylib.dir/build.make raylib/CMakeFiles/raylib.dir/build
make[2]: Entering directory '/Volumes/sensitive/raylib/build/linux'
[  1%] Building C object raylib/CMakeFiles/raylib.dir/rcore.c.o
cd /Volumes/sensitive/raylib/build/linux/raylib && /usr/bin/cc -DGRAPHICS_API_OPENGL_33 -I/Volumes/sensitive/raylib/src -fno-strict-aliasing -Werror=implicit-function-declaration -Werror=pointer-arith  -g -std=gnu99 -MD -MT raylib/CMakeFiles/raylib.dir/rcore.c.o -MF CMakeFiles/raylib.dir/rcore.c.o.d -o CMakeFiles/raylib.dir/rcore.c.o -c /Volumes/sensitive/raylib/src/rcore.c
/Volumes/sensitive/raylib/src/rcore.c: In function ‘InitGraphicsDevice’:
/Volumes/sensitive/raylib/src/rcore.c:4612:22: error: ‘eglGetProcAddress’ undeclared (first use in this function)
 4612 |     rlLoadExtensions(eglGetProcAddress);
      |                      ^~~~~~~~~~~~~~~~~
/Volumes/sensitive/raylib/src/rcore.c:4612:22: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [raylib/CMakeFiles/raylib.dir/build.make:76: raylib/CMakeFiles/raylib.dir/rcore.c.o] Error 1
make[2]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
make[1]: *** [CMakeFiles/Makefile2:362: raylib/CMakeFiles/raylib.dir/all] Error 2
make[1]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
make: *** [Makefile:166: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,18 @@
 cmake_minimum_required(VERSION 3.0)
 project(raylib)
 
+# Avoid excessive expansion of variables in conditionals. In particular, if
+# "PLATFORM" is "DRM" than:
+#
+# if (${PLATFORM} MATCHES "DRM")
+#
+# may expand e.g to:
+#
+# if (/usr/lib/aarch64-linux-gnu/libdrm.so MATCHES "DRM")
+#
+# See https://cmake.org/cmake/help/latest/policy/CMP0054.html
+cmake_policy(SET CMP0054 NEW)
+
 # Directory for easier includes
 # Anywhere you see include(...) you can check <root>/cmake for that file
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)

--- a/cmake/LibraryConfigurations.cmake
+++ b/cmake/LibraryConfigurations.cmake
@@ -1,6 +1,6 @@
 if (${PLATFORM} MATCHES "Desktop")
     set(PLATFORM_CPP "PLATFORM_DESKTOP")
-    
+
     if (APPLE)
         # Need to force OpenGL 3.3 on OS X
         # See: https://github.com/raysan5/raylib/issues/341
@@ -22,11 +22,11 @@ if (${PLATFORM} MATCHES "Desktop")
         if ("${OPENGL_LIBRARIES}" STREQUAL "")
             set(OPENGL_LIBRARIES "GL")
         endif ()
-        
+
         if ("${CMAKE_SYSTEM_NAME}" MATCHES "(Net|Open)BSD")
             find_library(OSS_LIBRARY ossaudio)
         endif ()
-        
+
         set(LIBS_PRIVATE m atomic pthread ${OPENGL_LIBRARIES} ${OSS_LIBRARY})
 
         if (USE_AUDIO)
@@ -48,16 +48,16 @@ elseif (${PLATFORM} MATCHES "Android")
     add_definitions(-DANDROID -D__ANDROID_API__=21)
     include_directories(${ANDROID_NDK}/sources/android/native_app_glue)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-libs,libatomic.a -Wl,--build-id -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--warn-shared-textrel -Wl,--fatal-warnings -u ANativeActivity_onCreate -Wl,-undefined,dynamic_lookup")
-    
+
     find_library(OPENGL_LIBRARY OpenGL)
     set(LIBS_PRIVATE m log android EGL GLESv2 OpenSLES atomic c)
 
 elseif (${PLATFORM} MATCHES "Raspberry Pi")
     set(PLATFORM_CPP "PLATFORM_RPI")
     set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
-    
+
     add_definitions(-D_DEFAULT_SOURCE)
-    
+
     find_library(GLESV2 brcmGLESv2 HINTS /opt/vc/lib)
     find_library(EGL brcmEGL HINTS /opt/vc/lib)
     find_library(BCMHOST bcm_host HINTS /opt/vc/lib)
@@ -65,19 +65,19 @@ elseif (${PLATFORM} MATCHES "Raspberry Pi")
     link_directories(/opt/vc/lib)
     set(LIBS_PRIVATE ${GLESV2} ${EGL} ${BCMHOST} pthread rt m dl)
 
-elseif (${PLATFORM} MATCHES "DRM")
+elseif ("${PLATFORM}" MATCHES "DRM")
     set(PLATFORM_CPP "PLATFORM_DRM")
     set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
-    
+
     add_definitions(-D_DEFAULT_SOURCE)
     add_definitions(-DEGL_NO_X11)
     add_definitions(-DPLATFORM_DRM)
-    
+
     find_library(GLESV2 GLESv2)
     find_library(EGL EGL)
     find_library(DRM drm)
     find_library(GBM gbm)
-    
+
     if (NOT CMAKE_CROSSCOMPILING)
         include_directories(/usr/include/libdrm)
     endif ()

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -52,7 +52,7 @@ foreach (example_dir ${example_dirs})
     # Get the .c files
     file(GLOB sources ${example_dir}/*.c)
     list(APPEND example_sources ${sources})
-    
+
     # Any any resources
     file(GLOB resources ${example_dir}/resources/*)
     list(APPEND example_resources ${resources})
@@ -71,7 +71,7 @@ if (${PLATFORM} MATCHES "Android")
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/core/core_3d_camera_free.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/core/core_3d_camera_first_person.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/core/core_world_screen.c)
-    
+
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_mesh_picking.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_cubicmap.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_skybox.c)
@@ -86,8 +86,8 @@ if (${PLATFORM} MATCHES "Android")
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_animation.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_first_person_maze.c)
 	list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/models/models_magicavoxel_loading.c)
-    
-	
+
+
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/shaders/shaders_custom_uniform.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/shaders/shaders_model_shader.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/shaders/shaders_postprocessing.c)
@@ -101,13 +101,13 @@ elseif (${PLATFORM} MATCHES "Web")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -s ALLOW_MEMORY_GROWTH=1 --no-heap-copy")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --shell-file ${CMAKE_SOURCE_DIR}/src/shell.html")
     set(CMAKE_EXECUTABLE_SUFFIX ".html")
-    
+
     # Remove the -rdynamic flag because otherwise emscripten
     # does not generate HTML+JS+WASM files, only a non-working
     # and fat HTML
     string(REPLACE "-rdynamic" "" CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS}")
 
-elseif (${PLATFORM} STREQUAL "DRM")
+elseif ("${PLATFORM}" STREQUAL "DRM")
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/others/rlgl_standalone.c)
     list(REMOVE_ITEM example_sources ${CMAKE_CURRENT_SOURCE_DIR}/others/raylib_opengl_interop.c)
 
@@ -124,15 +124,15 @@ foreach (example_source ${example_sources})
     # Create the basename for the example
     get_filename_component(example_name ${example_source} NAME)
     string(REPLACE ".c" "" example_name ${example_name})
-    
+
     # Setup the example
     add_executable(${example_name} ${example_source})
-    
+
     target_link_libraries(${example_name} raylib)
-    
+
     string(REGEX MATCH ".*/.*/" resources_dir ${example_source})
     string(APPEND resources_dir "resources")
-    
+
     if (${PLATFORM} MATCHES "Web" AND EXISTS ${resources_dir})
         # The local resources path needs to be mapped to /resources virtual path
         string(APPEND resources_dir "@resources")

